lib/engine/step/special_token.rb

```diff
@@ -31,27 +51,53 @@ def process_place_token(action)
           entity.owner,
           action.city,
           action.token,
-          teleport: ability(entity).teleport_price,
+          connected: false,
           special_ability: ability(entity),
         )
+
+        teleport_complete if @round.teleported
+      end
+
+      def process_pass(action)
+        @log << "#{action.entity.owner.name} (#{action.entity.sym}) declines to place token"
+        teleport_complete
+      end
+
+      def teleport_complete
+        ability = ability(@round.teleported)
+        @round.teleported.remove_ability(ability) if ability
+        @round.teleported = nil
       end
 
       def available_hex(entity, hex)
-        return if ability(entity).hexes.any? && !ability(entity).hexes.include?(hex.id)
+        return if !ability(entity).hexes.empty? && !ability(entity).hexes.include?(hex.id)
 
         @game.hex_by_id(hex.id).neighbors.keys
       end
 
       def available_tokens(entity)
-        return super unless ability(entity)&.extra
+        ability = ability(entity)
+        return super unless ability&.type == :token && ability.extra
 
         [Engine::Token.new(entity.owner)]
       end
 
+      def min_token_price(tokens)
+        return 0 if @round.teleported
+
+        super
+      end
+
       def ability(entity)
         return unless entity&.company?
 
-        @game.abilities(entity, :token)
+        @game.abilities(entity, :token) do |ability, _company|
+          return ability
+        end
+
+        @game.abilities(entity, :teleport) do |ability, _company|
+          return ability if ability.used?
+        end
```

Comment:
let's return nil explicitly at the end here just to be clear

Reply:
Fixed.